src/encoded/search/search.py

```
@@ -0,0 +1,1105 @@
+            'sort': {},
+            'clear_filters': self.clear_filters_setup()
+        }
+        self.add_search_header_if_needed()
+        self.set_pagination()
+        self.build_type_filters()
+
+    def list_source_fields(self):
+        """
+        Returns set of fields that are requested by user or default fields.
+        These fields are used to further limit the results from the search.
+        Note that you must provide the full fieldname with embeds, such as:
+        'field=biosample.biosource.individual.organism.name' and not just
+        'field=name'
+        """
+        fields_requested = self.request.normalized_params.getall('field')
+        if fields_requested:
+            fields = ['embedded.@id', 'embedded.@type']
+            for field in fields_requested:
+                fields.append('embedded.' + field)
+        elif self.search_frame in ['embedded', 'object', 'raw']:
+            if self.search_frame != 'embedded':
+                # frame=raw corresponds to 'properties' in ES
+                if self.search_frame == 'raw':
+                    frame = 'properties'
+                else:
+                    frame = self.search_frame
+                # let embedded be searched as well (for faceting)
+                fields = ['embedded.*', frame + '.*']
+            else:
+                fields = [self.search_frame + '.*']
+        else:
+            fields = ['embedded.*']
+        return fields
+
+    def build_query(self, source_fields):
+        """
+        Prepare the query within the Search object.
+        """
+        query_info = {}
+        string_query = None
+        # prepare the query from prepared_terms
+        for field, value in self.prepared_terms.items():
+            if field == 'q':
+                query_info['query'] = value
+                query_info['lenient'] = True
+                query_info['default_operator'] = 'AND'
+                query_info['fields'] = ['_all']
```

Is there a known default otherwise? Should the `query_info` be initialized to those defaults?

Can you add some text explaining how `q` interacts with other 'fields' such that it has such a dominant role over all other parts. A bit more overall documentation in the function doc string would be good. I feel like all this loop is doing is just:
```
if 'q' in self.prepared_terms:
    query_info[...] = ...
...
```
and moreover like this is just separating two different kinds of queries that should have their own functions.
It also seems like if there is a `q` you are ignoring other fields. Should you raise an error if those are there or do they come from an interface that can't help but send them?

---

I've simplified this method and clarified the docstring.